### DISCUSSION TITLE
Place to share presets?

### DISCUSSION THREAD

**Discussion by cdw2000** [ElectroTechnique/electrotechnique.github.io#12] [Software Issues]
Not really a software issue, but no other category choice.  I have a few presets I've created that others might enjoy.   And maybe others have created some as well that they would like to share.  Is it possible to setup a community presets area on here that users can upload to?   Should be pretty easy since presets are just text files.

**Comment by Hububalli** [comment]
Like the idea! How does it work with file names? Obviously it would be nice to not overwrite other presets and to also know where they came from.  Is it alphanumeric? I haven't tested it personally, only numerical.

**Reply by rolfdegen** [reply]
Saving and Naming Patches
Back button will cancel and move back to previous screen. If you make a mistake naming patches, press back button.
Save a new patch
Save 
Recall 
Turn the Recall encoder to select letters and numbers. Press Recall to select each character and move to the next one to select. Press to select final character.

Save
Overwriting an existing patch
Save 
Turn the Recall encoder to select the patch to overwrite.


**Reply by rolfdegen** [reply]
You can change and adding 999 Patches. The patch name is in the file.

https://i.ibb.co/xD0jJKX/SDCARD.jpg

**Reply by Hububalli** [reply]
As you say the patch names are in the file, I am curious about the actual file names. For example if I try out @cdw2000 's patches, and the file names are all just numbers, they will possibly overwrite some of mine. My question really was, are alphanumerical file names possible? 

Having CDW01, CDW02, CDW03 for example, to identify @cdw2000 's patches.

**Reply by cdw2000** [reply]
Alphanumeric patch filenames are not possible.  The numeric patch filename simply identifies which patch number gets assigned when scrolling through the patches on the TSynth.   Therefore, you could take my patch files and rename them to numbers that are not used in your collection and simply copy them to the MicroSD card.   When you power up the TSynth, my patches would show up in those numbered slots corresponding to their names.  The actual text name of the patch is in the file and will show on the display just like any others.

**Reply by cdw2000** [reply]
As a test here are 11 patches I created, numbered 61-71.   You can rename them to any unused numbers on your MicroSD card and then just copy them to the card in the same way you copied the initial EletroTechnique presets to the the card when you first set up your TSynth.
[CDW2000_patches.zip](https://github.com/ElectroTechnique/electrotechnique.github.io/files/5939842/CDW2000_patches.zip)

**Comment by Electrotechnique** [comment]
When I decided how the patch files would work, I thought that having them in an order might be useful, which is why the filenames are the order number. I wanted to avoid TSynth going through all the patches on the SD and having to decide what order to arrange them, at start-up. Whether this is the best system is debatable. Perhaps having arbitrary filenames and just loading them in whatever order it finds them on the card would be better. Having banks, which are basically folders with sets of patches in the them is another possibility, but then you need to program a suitable system to manage these from the encoder, buttons and display.

If someone wants to come up with a better way and code it, I'll include it in a release.

**Reply by rolfdegen** [reply]
In my last synth "Degenerator" I did that with 128 patch banks. The patch name was also in the file

Pic: https://drive.google.com/file/d/1nx8Ji2ULF-xY6yrVqUD6317ziRBY62fa/view?usp=sharing
Pic: https://drive.google.com/file/d/1JlQOlPl2WRrgZqvmto_MJvNFbfdkdB2f/view?usp=sharing

Greetings from germany. Rolf

**Comment by AnalogLabSwiss** [comment]
Hello, 

I created 8 new patches for the TSynth.
They can be downloaded for free at the following adress :https://synthpatch.ch/tsynth-patches.html

Numbers & names are : 
64 THE Bass
65 Oxy Wind
66 IronBass
67 Echoes
68 Wobbler
69 Hoover
70 Stargate
71 Stardust

Feel free to download !

All the best,

Carmelo
@analog_lab.swiss

**Reply by rolfdegen** [reply]
Doesn't it seem like it is free?

**Reply by Electrotechnique** [reply]
I think it's like _donationware_ and you can pay what you feel like, including zero.

**Reply by AnalogLabSwiss** [reply]
You can enter 0$ and download it for free. Or offer me a coffe. it's up to you ! More patches will come in the near futur.

Carmelo
IG : @analog_lab.swiss

**Comment by remip** [comment]
I made a website to preview the presets settings:

https://remip.github.io/tsynth-viewer/

**Reply by Hububalli** [reply]
Nice tool! This will be really useful.

**Reply by snowsh01** [reply]
Very nice. Maybe combine this with this...

http://lynx-tracks.com/apps/tsynth-patch-generator.php

**Comment by rolfdegen** [comment]
It's a very good idea :)

**Comment by snowsh01** [comment]
Random patch generator. Its a bit of a lucky dip, I had some trouble determining the min max values for each data part. If anyone has that let me know, I already spent too nuch time on this....

Feel free to try.

http://lynx-tracks.com/apps/tsynth-patch-generator.php
